storage/storage_image.go

```diff
@@ -427,23 +777,58 @@ func (s *storageImageDestination) SupportedManifestMIMETypes() []string {
 	return manifestMIMETypes
 }
 
-// PutManifest writes manifest to the destination.
-// FIXME? This should also receive a MIME type if known, to differentiate between schema versions.
-// If the destination is in principle available, refuses this manifest type (e.g. it does not recognize the schema),
-// but may accept a different manifest type, the returned error must be an ManifestTypeRejectedError.
+// GetManifest reads the manifest that we intend to store.  If we haven't been given one (yet?),
+// generate one.
+func (s *storageImageUnnamedDestination) GetManifest() ([]byte, string, error) {
+	if len(s.named.manifest) == 0 {
```

Comment:
There have been a few cases suggested to allow adding / modifying signatures in an `ImageDestination` without doing anything about the blobs/manifests… but that’s not quite relevant to this case.

In theory calling `PutBlob` is expected to also call `PutManifest`; if buildah or some other caller isn’t doing that currently, I guess I’d rather have them explicitly construct a manifest (using the discussed manifest interface I guess) than to have every destination deal with constructing a manifest from its own data, and do the otherwise often unnecessary bookkeeping to track all uploaded / reapplied blobs.